More randomized vendor stock

emmanuelbassil opened the pull request:
🆑 emmanuelbassil
tweak: Adds 'rare items' to sovietsoda, fitness, and games vendor.
bugfix: Fixes oversight allowing hacked vending machines to hurl traitor coin items at people.
/🆑 

I only did this for 'public' non-job essential vendors; I don't expect shortages to ever pop up this is only so the stock is less uniform which feels weird for vendors 'everyone' frequents. I did not do this for job essential vendors nor will I do it, randomness there is silly, I feel.

Also added rare items to the boda, fitness, games, and soda (this one not mapped currently) vendors. Moved out some contraband items --> rare in the boda and games, added new contraband to compensate that's more interesting. Also after some thought, will not add 'rare items' to any of the job essential vendors. Randomness on that front also seems silly. I feel it's important to make a few other vendors have 'rare' items and not only the existing soda one.

BurpleBineapple commented on code/modules/reagents/reagent_containers/drinkingglass/shaker.dm:
```diff
@@ -27,7 +27,4 @@
-	reagents.add_reagent(/datum/reagent/iron, 10)
-	reagents.add_reagent(/datum/reagent/nutriment/protein, 15)
-	reagents.add_reagent(/datum/reagent/water, 45)
+	reagents.add_reagent(/datum/reagent/drink/protein_shake, 100)
```
Not a fan of this change in particular. The protein shake is a mix of chems to get blood volume up and replenish hunger. Making it a unique reagent will only make it replenish your thirst.

BurpleBineapple replied:
Ideally you should make the new reagent do the same, or rename the specific flask item for protein shakes.

emmanuelbassil replied:
The new reagent should also replenish hunger

emmanuelbassil committed:
More random vendor items


Fixes


Auto stash before rebase of "refs/heads/RandVend"
Auto stash before rebase of "refs/heads/RandVend"
Auto stash before rebase of "refs/heads/RandVend"
Auto stash before rebase of "refs/heads/RandVend"
Auto stash before rebase of "refs/heads/RandVend"
Auto stash before rebase of "refs/heads/RandVend"
Auto stash before rebase of "refs/heads/RandVend"

emmanuelbassil commented:
Gave up on this protein shake thing for now.
Chemistry code is full of terrors

MuckerMayhem commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Drinks.dm:
```diff
@@ -244,7 +244,7 @@
 	glass_name = "th'oom juice"
 	glass_desc = "sweet and savory goodness!"
-	sugar_amount = 50
+	sugar_amount = 0.5
```
This doesn't seem to be documented, is there a reason this was changed so drastically?
